### ISSUE TITLE
Could not load file or assembly WebActivatorEx

### ISSUE THREAD

**Issue by kipusoep** [jetheredge/SquishIt#327]
Installing SquishIt version 0.9.8.1 via NuGet gives the following exception at runtime:

<pre>Could not load file or assembly WebActivatorEx, Version=2.0.0.0, Culture=neutral, PublicKeyToken=7b26dc2a43f6a0d4 or one of its dependencies. The system cannot find the file specified.</pre>

The dependency WebActivator (≥ 1.4.0) as shown on ([this page](https://www.nuget.org/packages/SquishIt.Less/)) isn't there in NuGet's UI and thus not added.

**Comment by AlexCuse** [comment]
Ugh.  The original WebActivator package was de-listed.  I will need to test with v2 and make sure everything works, i'll try to get 0.9.8.2 out soon I guess.  Kind of frustrating that it won't allow installation once it is de-listed, wonder how many other projects that breaks.

**Comment by AlexCuse** [comment]
If you install WebActivatorEx v2.0 does it work?

**Comment by kipusoep** [comment]
Sorry, didn't check that, reverted to previous SquishIt right away.

**Comment by AlexCuse** [comment]
So there are no hard references to WebActivator anywhere (there will be in v1.0)

Guessing it is probably related to this file that gets transformed by nuget: https://github.com/jetheredge/SquishIt/blob/a00720b09fdd42920aa723944cb2ae1f58507c72/SquishIt.Less/Content/App_Start/SquishItLess.cs.pp

You reverted via git checkout (or similar) right?  I would expect the problem to occur regardless of version if you install the preprocessor package after the date WebActivator was delisted.  

So if you install/update a preprocessor and experience this problem, I'm guessing the fix will be to install WebActivatorEx on its own.  I will update it in future packages once I am done cursing nuget for this.

**Comment by kipusoep** [comment]
I reverted by installing the previous SquishIt version on NuGet.
The reason this exception occurred is because of Umbraco (CMS). The Boot manager of Umbraco registers plug-ins and scans all assemblies in the AppDomain. During this scan any non-resolvable references throw an exception.

**Comment by AlexCuse** [comment]
Huh, I wouldn't expect the previous version to work either.  You are using SquishIt.Less right?  It has always referenced WebActivator (not EX).

Anyway you can try this: https://www.nuget.org/packages/SquishIt.Less/0.9.8.2-beta1

**Comment by kipusoep** [comment]
No I'm talking about the regular SquishIt package. I'm sure it had a
dependency to WebActivator :-/


Op 5 jan. 2016 om 23:26 heeft Alex Ullrich <notifications@github.com> het
volgende geschreven:

Huh, I wouldn't expect the previous version to work either. You are using
SquishIt.Less right? It has always referenced WebActivator (not EX).

Anyway you can try this:
https://www.nuget.org/packages/SquishIt.Less/0.9.8.2-beta1

—
Reply to this email directly or view it on GitHub
<https://github.com/jetheredge/SquishIt/issues/327#issuecomment-169153871>.

**Comment by AlexCuse** [comment]
Nah, it shouldn't.  v1.0 will, but 0.9.8.1 doesn't have any dependency on WebActivator (hard or soft).

I'll try to look into this some more tomorrow but I don't see where that message would've come from unless you were using a preprocessor.

**Comment by kipusoep** [comment]
Sorry for the large screenshot and Dutch language ;-)

![webactivatorex-exception](https://cloud.githubusercontent.com/assets/1866549/12138207/69af0f58-b457-11e5-81ee-6e30771a3b0f.png)

Opening the assembly in ILSpy:

![webactivatorex-reference](https://cloud.githubusercontent.com/assets/1866549/12138235/a1e62924-b457-11e5-9932-6c19ae3d71a4.png)

**Comment by AlexCuse** [comment]
This is weird, looks like a 1.0 assembly got pushed with the old package definition.  Thanks for pointing this out - I pulled 0.9.8.1 and am working on resolution now.

**Comment by AlexCuse** [comment]
Just pushed 0.9.8.2, it should solve the problem.  Thanks again for pointing this out!

![image](https://cloud.githubusercontent.com/assets/194594/12145886/c10fbd5e-b45e-11e5-9280-616093a590f4.png)
